docs/CONTRIBUTING.md

```diff
@@ -7,66 +7,154 @@ helps, and credit will always be given.
 
-    ```bash
-    make build
-    ```
+        make build
 
     If you have any AWS credentials configured and would like to run the full
     remote IO test suite:
 
-    ```bash
-    make build-with-remote
-    ```
+        make build-with-remote
+
+8.  Commit your changes and push your branch to GitHub:
+
+        git add .
+        git commit -m "Resolves gh-###. Your detailed description of your changes."
+        git push origin {your_development_type}/short-description
+
+9.  Submit a pull request through the GitHub website.
+
+## Adding a New Custom Reader
+
+### Basics
+
+AICSImageIO is broken into the `Reader` base class and the `AICSImage` higher
+level "standardization" class. `Reader` classes should always provide
+access in some sense to the "raw" data. The `AICSImage` class wraps the
+`Reader` classes to then select and transform the "raw" data into our
+AICS / near-OME standard model.
+
+### Reader Class Implementation
+
+The `Reader` base class can largely be thought of as a wrapper around
+`xarray` due to how we largely store most or all data and metadata on
+`xarray.DataArray` objects.
+Because of this, it may be useful to read the
+[xarray documentation](http://xarray.pydata.org/en/stable/).
+
+The `Reader` base class in full can be found
+[here](https://github.com/AllenCellModeling/aicsimageio/blob/main/aicsimageio/readers/reader.py).
+
+New `Reader` classes are **required** to implement:
+
+-   `_is_supported_image`: a function to check if the custom `Reader` class
+    _can_ read the provided object (file, array, etc.)
+-   `scenes`: a property which returns the Tuple of all scene names
+    (or in OME terms, "images" or "series").
+-   `_read_delayed`: a function which returns an `xarray.DataArray` object
+    which is backed by a delayed `dask.array.Array`.
+-   `_read_immediate`: a function which returns an `xarray.DataArray` object
+    which is backed by an in-memory `numpy.ndarray`.
+
+New `Reader` classes can **optionally** implement:
+
+-   `_get_stitched_dask_mosaic`: otherwises raises `NotImplementedError`
+-   `_get_stitched_mosaic`: otherwises raises `NotImplementedError`
+-   `get_mosaic_tile_position`: otherwises raises `NotImplementedError`
+-   `ome_metadata`: otherwises raises `NotImplementedError`
+-   `physical_pixel_sizes`: default returns `(None, None, None)`
+
+Please see `Reader` code for docstrings on each of these
+optional functions or properties.
+
+After the **required** functions and properties have been implemented,
+`Reader` class functions like `get_image_data` and properties like `channel_names`
+are then handled for you because the `Reader` base class has built-in functions to
+operate against the provided / stored xarray objects.
+
+### Custom Dependencies
+
+If your `Reader` requires custom dependencies, add the custom dependencies
+to our [format_libs lookup in setup.py](https://github.com/AllenCellModeling/aicsimageio/blob/main/setup.py#L26).
```

Comment:
```suggestion
to our [`format_libs` lookup in `setup.py`](https://github.com/AllenCellModeling/aicsimageio/blob/main/setup.py#L26).
```

Reply:
Can't do backticks in links... i.e. ```[`something`]```, it breaks the doc renderer :slightly_frowning_face:

Reply:
Ah drat :disappointed: that works on GitHub's markdown renderer :woman_shrugging: